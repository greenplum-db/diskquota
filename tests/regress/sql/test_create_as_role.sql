CREATE ROLE r;
SELECT diskquota.set_role_quota('r', '10MB');
SET ROLE r;
select pg_sleep(5);

-- heap table
CREATE TABLE t1 AS SELECT generate_series(1, 100000000);
select pg_sleep(5);

-- temp table
CREATE TEMP TABLE t2 AS SELECT generate_series(1, 100000000);
select pg_sleep(5);

-- toast table
CREATE TABLE toast_table AS SELECT array(select * from generate_series(1,10000)) FROM generate_series(1, 100000);
select pg_sleep(5);

-- ao table
CREATE TABLE ao_table WITH(appendonly=true) AS SELECT generate_series(1, 100000000);
select pg_sleep(5);

-- aocs table
CREATE TABLE aocs_table WITH(appendonly=true, orientation=column) AS SELECT i, array(select * from generate_series(1,10000)) FROM generate_series(1, 100000) AS i;
select pg_sleep(5);

DROP TABLE IF EXISTS t1;
DROP TABLE IF EXISTS t2;
DROP TABLE IF EXISTS toast_table;
DROP TABLE IF EXISTS ao_table;
DROP TABLE IF EXISTS aocs_table;
RESET ROLE;
DROP ROLE r;