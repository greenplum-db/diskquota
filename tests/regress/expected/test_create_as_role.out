CREATE ROLE r;
NOTICE:  resource queue required -- using default resource queue "pg_default"
SELECT diskquota.set_role_quota('r', '10MB');
 set_role_quota 
----------------
 
(1 row)

SET ROLE r;
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- heap table
CREATE TABLE t1 AS SELECT generate_series(1, 100000000);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause. Creating a NULL policy entry.
ERROR:  role's disk space quota exceeded with name:37583  (seg0 127.0.0.1:6002 pid=34947)
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- temp table
CREATE TEMP TABLE t2 AS SELECT generate_series(1, 100000000);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause. Creating a NULL policy entry.
ERROR:  role's disk space quota exceeded with name:37583  (seg1 127.0.0.1:6003 pid=34945)
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- toast table
CREATE TABLE toast_table AS SELECT array(select * from generate_series(1,10000)) FROM generate_series(1, 100000);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'array' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ERROR:  role's disk space quota exceeded with name:37583  (seg1 127.0.0.1:6003 pid=34945)
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- ao table
CREATE TABLE ao_table WITH(appendonly=true) AS SELECT generate_series(1, 100000000);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause. Creating a NULL policy entry.
ERROR:  role's disk space quota exceeded with name:37583  (seg1 127.0.0.1:6003 pid=34945)
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- aocs table
CREATE TABLE aocs_table WITH(appendonly=true, orientation=column) AS SELECT i, array(select * from generate_series(1,10000)) FROM generate_series(1, 100000) AS i;
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ERROR:  role's disk space quota exceeded with name:37583  (seg2 127.0.0.1:6004 pid=34946)
select pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

DROP TABLE IF EXISTS t1;
NOTICE:  table "t1" does not exist, skipping
DROP TABLE IF EXISTS t2;
NOTICE:  table "t2" does not exist, skipping
DROP TABLE IF EXISTS toast_table;
NOTICE:  table "toast_table" does not exist, skipping
DROP TABLE IF EXISTS ao_table;
NOTICE:  table "ao_table" does not exist, skipping
DROP TABLE IF EXISTS aocs_table;
NOTICE:  table "aocs_table" does not exist, skipping
RESET ROLE;
DROP ROLE r;
